src/cacheSeed/cacheSeedManager.ts

```diff
@@ -0,0 +1,75 @@
+import { Logger } from '@map-colonies/js-logger';
+import { inject, singleton } from 'tsyringe';
+import { OperationStatus } from '@map-colonies/mc-priority-queue';
+import { SERVICES } from '../common/constants';
+import { IConfig, IJobParams, ISeedOptions, ITaskParams } from '../common/interfaces';
+import { MapproxySeed } from '../mapproxyUtils/mapproxySeed';
+import { MapproxyConfigClient } from '../clients/mapproxyConfig';
+import { QueueClient } from '../clients/queueClient';
+import { SeedMode } from '../common/enums';
+
+@singleton()
+export class CacheSeedManager {
+  private readonly seedAttempts: number;
+  private readonly taskType: string;
+
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    private readonly queueClient: QueueClient,
+    private readonly mapproxyConfigClient: MapproxyConfigClient,
+    private readonly mapproxySeed: MapproxySeed
+  ) {
+    this.seedAttempts = this.config.get<number>('seedAttempts');
+    this.taskType = this.config.get<string>('queue.tilesTaskType');
+  }
+
+  public async handleCacheSeedTask(): Promise<boolean> {
+    this.logger.info('Running search for seed jobs');
+    const tilesTask = await this.queueClient.queueHandlerForTileSeedingTasks.dequeue<ITaskParams>(this.taskType);
+    if (tilesTask) {
+      const job = await this.queueClient.jobsClient.getJob<IJobParams, ITaskParams>(tilesTask.jobId);
+      const jobId = tilesTask.jobId;
+      const taskId = tilesTask.id;
+      const attempts = tilesTask.attempts;
+      const seeds = tilesTask.parameters.seedTasks;
+      this.logger.info({
+        msg: `Found new seed job: ${jobId}`,
+        jobId,
+        productId: job.resourceId,
+        productVersion: job.version,
+        productType: job.productType,
+      });
+
+      if (attempts <= this.seedAttempts) {
+        try {
+          await this.runTask(seeds);
+          await this.queueClient.queueHandlerForTileSeedingTasks.ack(jobId, taskId);
+          await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, {
+            status: OperationStatus.COMPLETED,
+            percentage: 100,
+          });
+        } catch (error) {
+          await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, true, (error as Error).message);
+        }
+      } else {
+        await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, false);
+        await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, { status: OperationStatus.FAILED });
+      }
+    }
+    return Boolean(tilesTask);
+  }
+
+  private async runTask(seedTasks: ISeedOptions[]): Promise<void> {
+    const currentMapproxyConfig = await this.mapproxyConfigClient.getConfig(); // todo - will be used on seeding functions
```

Comment:
please get configs in ctor

Reply:
this is mapproxy.yaml current configuration and should be loaded on each seed iteration.
the ctor injected as part as main worker's handler